### ISSUE TITLE
Add some of the other `cabal-fmt` refactorings?

### ISSUE THREAD

**Issue by michaelpj** [tfausak/cabal-gild#94]
`cabal-fmt` has, as far as I know, two additional undocumented "refactorings" beyond expanding modules:
1. Inserting "fragments", useful for sharing things between cabal files, e.g. https://github.com/phadej/cabal-fmt/blob/master/fixtures/fragment-section.cabal
2. Expanding file globs in some fields, e.g. https://github.com/phadej/cabal-fmt/blob/master/fixtures/simple-example.cabal#L43

I think both of these are quite nice features, it would be cool if `cabal-gild` had something similar.

**Comment by tfausak** [comment]
`cabal-fmt`'s `glob-files` pragma should be supported by Gild's `discover` pragma since #86. It uses the same syntax as the module discovery pragma but it discovers files instead of modules. 

I would like to add support for importing fragments.

**Comment by tfausak** [comment]
I don't see any other pragmas that could be ported over: https://github.com/phadej/cabal-fmt/issues/35

https://github.com/phadej/cabal-fmt/blob/86d59df7579cdf97b8169fa37aef7dda716abad2/src/CabalFmt/Pragma.hs#L58-L63

**Comment by michaelpj** [comment]
Yeah, I guess the fragments is the main one. This came up because we're thinking of using cabal-fmt's fragments, and I would otherwise suggest using cabal-gild :)

**Comment by tfausak** [comment]
I've tagged this as "help wanted". It would be nice for Gild to support importing fragments. I doubt I'll have the time to add that feature myself.

**Comment by BebeSparkelSparkel** [comment]
@michaelpj @michaelpj Have you worked on this at all? If so, could you share your changes in a draft pr?

**Comment by michaelpj** [comment]
I have not looked at this.

**Comment by tfausak** [comment]
No, I have not yet looked at this. I doubt I will anytime soon. 

My only thought about the implementation is that fragments importing other fragments is perhaps desirable but also could get stuck in a loop. Perhaps there should be a configurable maximum depth that defaults to something reasonable.
